src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const HALF_HEALTH = 50;
+const DEAD = 0;
+
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = FULL_HEALTH;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if ((animal instanceof Herbivore) && (animal.hidden === false)) {
+      animal.health -= HALF_HEALTH;
+
+      Animal.alive = Animal.alive.filter(({ health }) => health > DEAD);
```

Comment:
You don't need to do useless filtering if `animal.health > DEAD`. You can add 1 more condition for this